### ISSUE TITLE
Crash when switching to backgroundplayer

### ISSUE THREAD

**Issue by paul70078** [TeamNewPipe/NewPipe#1920]
When playing a Video in Fullscreen (problem existed with several Videos with different length) and switching to the background player the App crashes.

It is kind of similar to #1779, but the app doesnt freeze as long and the stackdump reports another error

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ AT
* __Service:__ none
* __Version:__ 0.14.2
* __OS:__ Linux Android 8.0.0 - 26

<details><summary><b>Crash log</b></summary><p>

```
java.lang.RuntimeException: Unable to stop activity {org.schabi.newpipe/org.schabi.newpipe.player.MainVideoPlayer}: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4998)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5055)
	at android.app.ActivityThread.-wrap6(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1945)
	at android.os.Handler.dispatchMessage(Handler.java:108)
	at android.os.Looper.loop(Looper.java:166)
	at android.app.ActivityThread.main(ActivityThread.java:7415)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference
	at org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)
	at org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)
	at org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)
	at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1386)
	at android.app.Activity.performStop(Activity.java:7578)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4993)
	... 9 more
java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference
	at org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)
	at org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)
	at org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)
	at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1386)
	at android.app.Activity.performStop(Activity.java:7578)
	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4993)
	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5055)
	at android.app.ActivityThread.-wrap6(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1945)
	at android.os.Handler.dispatchMessage(Handler.java:108)
	at android.os.Looper.loop(Looper.java:166)
	at android.app.ActivityThread.main(ActivityThread.java:7415)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)

```
</p></details>
<hr>

**Comment by hueldoeu** [comment]
0.14.1 is way betterthan 0.14.2

Popup player is good but doesn't work in 0.14.2 don't know why

**Comment by catfluoride** [comment]
Can confirm. Here it goes my crashlog:
`{
   "user_action": "ui error",
   "request": "App crash, UI failure",
   "content_language": "US",
   "service": "none",
   "package": "org.schabi.newpipe",
   "version": "0.14.2",
   "os": "Linux HONOR\/STF-L09\/HWSTF:8.0.0\/HUAWEISTF-L09\/376(C432):user\/release-keys 8.0.0 - 26",
   "time": "2018-12-03 08:55",
   "exceptions": [
      "java.lang.RuntimeException: Unable to stop activity {org.schabi.newpipe\/org.schabi.newpipe.player.MainVideoPlayer}: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference\n\tat android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5111)\n\tat android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5168)\n\tat android.app.ActivityThread.-wrap6(Unknown Source:0)\n\tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:2058)\n\tat android.os.Handler.dispatchMessage(Handler.java:108)\n\tat android.os.Looper.loop(Looper.java:166)\n\tat android.app.ActivityThread.main(ActivityThread.java:7523)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)\nCaused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference\n\tat org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)\n\tat org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)\n\tat org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)\n\tat android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1386)\n\tat android.app.Activity.performStop(Activity.java:7587)\n\tat android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5106)\n\t... 9 more\njava.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference\n\tat org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)\n\tat org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)\n\tat org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)\n\tat android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1386)\n\tat android.app.Activity.performStop(Activity.java:7587)\n\tat android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5106)\n\tat android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5168)\n\tat android.app.ActivityThread.-wrap6(Unknown Source:0)\n\tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:2058)\n\tat android.os.Handler.dispatchMessage(Handler.java:108)\n\tat android.os.Looper.loop(Looper.java:166)\n\tat android.app.ActivityThread.main(ActivityThread.java:7523)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)\n"
   ],
   "user_comment": ""
}`

**Comment by francoism90** [comment]
This issue can be solved by building a (debug) APK with Android Studio. It would be better to have a new APK release on F-droid, but at the moment this seems to be the only solution.

**Comment by shadow303** [comment]
Same error. UI crash from full screen player going to either Background player or going to the floating window

{
   "user_action": "ui error",
   "request": "App crash, UI failure",
   "content_language": "US",
   "service": "none",
   "package": "org.schabi.newpipe",
   "version": "0.14.2",
   "os": "Linux Android 7.0 - 24",
   "time": "2018-12-03 17:24",
   "exceptions": [
      "java.lang.RuntimeException: Unable to stop activity {org.schabi.newpipe\/org.schabi.newpipe.player.MainVideoPlayer}: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference\n\tat android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4272)\n\tat android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4328)\n\tat android.app.ActivityThread.-wrap6(ActivityThread.java)\n\tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:1604)\n\tat android.os.Handler.dispatchMessage(Handler.java:102)\n\tat android.os.Looper.loop(Looper.java:165)\n\tat android.app.ActivityThread.main(ActivityThread.java:6375)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:883)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:773)\nCaused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference\n\tat org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)\n\tat org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)\n\tat org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)\n\tat android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1290)\n\tat android.app.Activity.performStop(Activity.java:7029)\n\tat android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4267)\n\t... 9 more\njava.lang.NullPointerException: Attempt to invoke virtual method 'int com.google.android.exoplayer2.SimpleExoPlayer.getPlaybackState()' on a null object reference\n\tat org.schabi.newpipe.player.BasePlayer.isPlaying(BasePlayer.java:1109)\n\tat org.schabi.newpipe.player.MainVideoPlayer.createPlayerState(MainVideoPlayer.java:252)\n\tat org.schabi.newpipe.player.MainVideoPlayer.onStop(MainVideoPlayer.java:237)\n\tat android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1290)\n\tat android.app.Activity.performStop(Activity.java:7029)\n\tat android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4267)\n\tat android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4328)\n\tat android.app.ActivityThread.-wrap6(ActivityThread.java)\n\tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:1604)\n\tat android.os.Handler.dispatchMessage(Handler.java:102)\n\tat android.os.Looper.loop(Looper.java:165)\n\tat android.app.ActivityThread.main(ActivityThread.java:6375)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:883)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:773)\n"
   ],
   "user_comment": ""
}

**Comment by c-e-p-x-u-o** [comment]
same issue

**Comment by dpuza** [comment]
+1

**Comment by TobiGr** [comment]
Fixed in #1874 and #1902

**Comment by dpuza** [comment]
@tobigr when would these be released next?  I uninstalled and installed 14.1.
